hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,28 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    String database = identifier.namespace().level(0);
+    try {
+      Table table = clients.run(client -> client.getTable(database, identifier.name()));
+      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, identifier));
```

Comment:
Also here, should we return false rather than throw exception?  (what was the original behavior?)

Reply:
Well, I guess the original throws exception?

Reply:
and add a test to be explicit :)